docs/GDPs/gdp-duck-storage.rst

```diff
@@ -0,0 +1,456 @@
+:code:`sync_state`:
+   gt4py.storage.SyncState, supported by the :code:`asstorage` and :code:`storage` functions,  only has effect if
+   :code:`managed="gt4py"`. indicates which of the provided buffers (among :code:`data`, :code:`device_data`) is up to
+   date at the time of initialization.
+:code:`copy`
+   Supported by the :code:`storage` function. It can be used to specify whether the value given by :code:`data` or
+   :code:`device_data` is copied or not.
+
+If a parameter is not explicitly specified, it is inferred from the default parameter set. If there is no default
+parameter set provided or it does not provide the required information, it is gathered from the :code:`data` or
+:code:`device_data` parameters. If this does not provide this information, a trivial default value is assumed. If no
+default value is available, an error is raised that the parameters are underdetermined.
+
+If :code:`copy=False` and neither :code:`data` nor :code:`device_data` are provided, the other arguments are used to
+allocate an appropriate buffer. If :code:`data` or :code:`device_data` is provided, the consistency of the parameters
+with the buffers is validated.
+
+If the field is not 3-D, as indicated by :code:`axes`, the length of parameters :code:`default_origin` and
+:code:`shape`, may either be of length 3 or of the actual dimension of the storage.
+
+We further expose the :code:`Storage` base class, mainly to enable type checking. It can alternatively be used in the
+same way as :code:`storage` to initialize storages. On the other hand, constructors of the derived, hardware-specific
+storage types (See Section :ref:`storage_types`) are not intended to be used directly.
+
+
+Storage Attributes
+^^^^^^^^^^^^^^^^^^
+
+While we aim at supporting as many features as possible, we have not compiled an exhaustive list of features yet and we
+expressly ask for suggestions here (focusing on NumPy functions of the form :code:`np.function` or attributes and
+methods of ndarrays of the form :code:`ndarray.attribute` or :code:`ndarray.method()`.)
+
+Supported numpy functions:
+
+:code:`np.all`, :code:`np.any`
+   same semantics as :code:`np.logical_and.reduce` and :code:`np.logical_or.reduce`, respectively
+
+Features not supported due to unclear semantics:
+
+:code:`transpose`
+   It does not seem to make sense to swap spatial dimensions
+
+.. _constructors:
+
+Properties
+==========
+
+:code:`ndims`
+   number of (unmasked) dimensions
+:code:`shape`
+    tuple of length :code:`ndims`,
+:code:`strides`
+    tuple of length :code:`ndims`,
+:code:`data`
+   returns :code:`data` attribute of the underlying ndarray
+:code:`alignment`
+   the value given in the constructor
+:code:`axes`
+   string of unmasked axes, e.g. :code:`"IJ"` for a 2d field spanning longitude and latitude but not the vertical.
+:code:`mask`
+   similar to axes, but a tuple of booleans. :code:`(True, True, False)` would be a 2d field spanning longitude and
```

Comment:
It's there for convenience , since it often leads to to nicer code than to actually look up which axes are present in `axes` each time. can also remove it, and only use `_mask` internally.